quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/services/QuarkusValidator.java

```diff
@@ -130,13 +130,61 @@ private void validatePropertyValue(String propertyName, ExtendedConfigDescriptio
 			return;
 		}
 
-		// TODO validate boolean, int, enums types
-		String type = metadata.getType();
+		String value = property.getValue().getText().trim();
+		if (value.length() == 0) {
+			return;
+		}
+		
+		String errorMessage = null;
+		if (isInvalidEnumValue(metadata, value)) {
+			errorMessage = "Invalid enum value: '" + property.getValue().getText().trim() + "' is invalid for type " + metadata.getType();
+		} else if (isValueTypeMismatch(metadata, value)) {
+			errorMessage = "Type mismatch: " + metadata.getType() + " expected";
+		}
+
+		if (errorMessage != null) {
+			addDiagnostic(errorMessage, property.getValue(), severity, ValidationType.value.name());
+		}
+	}
+
+	private boolean isInvalidEnumValue(ExtendedConfigDescriptionBuildItem metadata, String value) {

```

Comment:
it's an utility method, please use static

Reply:
just to confirm, should it be public static or private static?

Reply:
private static as it is used only in this code.  If we see that this method will be used in the future in another code, we could create an utility class with static public.